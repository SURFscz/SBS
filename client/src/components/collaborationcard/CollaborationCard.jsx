import React from "react";

import "./CollaborationCard.scss";
import Logo from "../_redesign/logo/Logo";
import {Button, Chip, ChipType} from "@surfnet/sds";
import I18n from "../../locale/I18n";

export default function CollaborationCard({
                                              collaboration,
                                              user,
                                              history
                                          }) {

    const openCollaboration = () => {
        history.push(`/collaborations/${collaboration.id}`);
    }


    const isAdmin = (user.collaboration_memberships
        .find(cm => cm.collaboration_id === collaboration.id) || {}).role === "admin";
    return (
        <div key={collaboration.id} className="sds--content-card">
            <div className="sds--content-card--main">
                <div className="sds--content-card--visual">
                    <Logo src={collaboration.logo}/>
                </div>
                <div className="sds--content-card--textual">
                    <div className="sds--content-card--text-and-actions">
                        <div>
                            <p className="org-name">{collaboration.organisation.name}</p>
                            <p className="sds--space--bottom--1 coll-name">
                                {collaboration.name}
                            </p>
                        </div>
                        <div className="sds--content-card--actions">
                            {isAdmin && <Chip label={I18n.t("profile.admin")} type={ChipType.Main_400}/>}
                            <Button txt={I18n.t("collaborationsOverview.open")}
                                    onClick={openCollaboration}/>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    )
}
