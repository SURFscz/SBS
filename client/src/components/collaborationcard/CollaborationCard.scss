@import "../../stylesheets/vars";

.sds--content-card {
    a.more-link {
        display: flex;

        svg {
            width: 16px;
            height: auto;
            margin-left: 8px;
        }
    }

    p.org-name {
        color: var(--sds--color--gray--500)
    }
    p.coll-name {
        font-weight: 600;
        font-size: 18px;
    }

    .sds--content-card--visual {
        img {
            width: 90px;
            height: auto;
        }
    }

}
