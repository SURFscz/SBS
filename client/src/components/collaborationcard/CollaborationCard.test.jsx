import React from "react";
import {render} from "@testing-library/react";
import CollaborationCard from "./CollaborationCard";

jest.mock("../../locale/I18n", () => ({
    t: key => key
}));
jest.mock("../_redesign/logo/Logo", () => props => (
    <img data-testid="logo" {...props} />
));
jest.mock("@fortawesome/react-fontawesome", () => ({
    FontAwesomeIcon: () => <span data-testid="icon"/>
}));

const collaboration = {
    id: 1,
    name: "Test Collaboration",
    description: "A test collaboration",
    logo: "logo.png",
    organisation: {name: "Test Org"}
};

const user = {
    collaboration_memberships: [
        {collaboration_id: 1, role: "admin"}
    ]
};

const history = {push: jest.fn()};

describe("CollaborationCard", () => {
    it("matches snapshot", () => {
        const {asFragment} = render(
            <CollaborationCard
                collaboration={collaboration}
                user={user}
                history={history}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
