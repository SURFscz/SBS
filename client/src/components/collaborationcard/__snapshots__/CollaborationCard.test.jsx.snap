// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollaborationCard matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sds--content-card"
  >
    <div
      class="sds--content-card--main"
    >
      <div
        class="sds--content-card--visual"
      >
        <img
          data-testid="logo"
          src="logo.png"
        />
      </div>
      <div
        class="sds--content-card--textual"
      >
        <div
          class="sds--content-card--text-and-actions"
        >
          <div>
            <p
              class="org-name"
            >
              Test Org
            </p>
            <p
              class="sds--space--bottom--1 coll-name"
            >
              Test Collaboration
            </p>
          </div>
          <div
            class="sds--content-card--actions"
          >
            <button
              class="sds--chips main-400"
            >
              <span
                class="sds--chips--textual sds--text--body--small"
              >
                profile.admin
              </span>
            </button>
            <button
              class="sds--btn sds--btn--primary     "
              type="button"
            >
              <span
                class="textual"
              >
                collaborationsOverview.open
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
